chalice/awsclient.py

```diff
@@ -812,7 +838,7 @@ def _add_lambda_permission_if_needed(self, source_arn, function_arn,
         policy = self.get_function_policy(function_arn)
         if self._policy_gives_access(policy, source_arn, service_name):
             return
-        random_id = self._random_id()
+        random_id = generate_random_id()
```

Comment:
Let's keep this an internal method unless some other module uses the utility function and find some other way to get the file under 1000 lines.